packages/scaffold-ui/src/modal/w3m-modal/index.ts

```diff
@@ -189,31 +189,37 @@ export class W3mModal extends LitElement {
   }
 
   private async onNewAddress(caipAddress?: CaipAddress) {
-    const prevConnected = this.caipAddress
-      ? CoreHelperUtil.getPlainAddress(this.caipAddress)
+    const prevCaipAddress = this.caipAddress
+    const prevConnected = prevCaipAddress
+      ? CoreHelperUtil.getPlainAddress(prevCaipAddress)
       : undefined
     const nextConnected = caipAddress ? CoreHelperUtil.getPlainAddress(caipAddress) : undefined
     const isSameAddress = prevConnected === nextConnected
 
+    this.caipAddress = caipAddress
+
     if (nextConnected && !isSameAddress && this.isSiweEnabled) {
```

Comment:
There were some asynchronous tasks happening before setting caipAddress. In here i just made sure to assign caipAddress before asynchronous tasks start to kick in.

Reply:
What was the async call? @magiziz

Reply:
It this the one fixing the main issue?

Reply:
Yeah